src/api/uploads/FolderUploadNode.js

```diff
@@ -117,18 +117,20 @@ class FolderUploadNode {
             const data = await this.createFolder();
             this.folderId = data.id;
         } catch (error) {
-            errorEncountered = true;
-            errorCode = error.code;
             // @TODO: Handle 429
             if (error.code === ERROR_CODE_ITEM_NAME_IN_USE) {
                 this.folderId = error.context_info.conflicts[0].id;
             } else if (isRoot) {
+                errorEncountered = true;
+                errorCode = error.code;
                 errorCallback(error);
             } else {
                 // If this is a child folder of the folder being uploaded, this errorCallback will set
                 // an error message on the root folder being uploaded. Set a generic messages saying that a
                 // child has caused the error. The child folder will be tagged with the error message in
                 // the call to this.addFolderToUploadQueue below
+                errorEncountered = true;
+                errorCode = error.code;
```

Comment:
This variable doesn't seem particularly useful. Can we use `error.code` directly on line 154?

Reply:
probably so. i was simply moving the error var.s to each block and removing the one that should not be set, so I left as much of the logic as it was before

Reply:
oh actually, we still need `errorCode` as `error` only exists in the catch block